pkg/rid/store/cockroach/identification_service_area.go

```diff
@@ -222,3 +222,25 @@ func (c *isaRepo) SearchISAs(ctx context.Context, cells s2.CellUnion, earliest *
 
 	return c.process(ctx, isasInCellsQuery, earliest, latest, pq.Int64Array(cids))
 }
+
+// ListExpiredISAs lists all expired ISAs based and writer.
+// Records expire if current time is <expiredDurationInMin> minutes more than records' endTime.
+func (c *isaRepo) ListExpiredISAs(ctx context.Context, writer string) ([]*ridmodels.IdentificationServiceArea, error) {
+	var (
+		isasInCellsQuery = fmt.Sprintf(`
+	SELECT
+		%s
+	FROM
+		identification_service_areas
+	WHERE
+		ends_at + INTERVAL '%d' MINUTE <= CURRENT_TIMESTAMP
+	AND
+		writer = $1`, isaFields, expiredDurationInMin)
+	)
+
+	if len(writer) == 0 {
+		return nil, stacktrace.NewError("Writer is required")
```

Comment:
How will ISAs with no writer be removed if we can't specify an empty string here?

Reply:
removed condition and add ListExpiredSubscriptionsWithNullWriter function to remove expired record with writer = NULL (writer can be NULL when db is migrated from v3.0 to v3.1)